cmd/wavefront-collector/main.go

```diff
@@ -112,7 +112,7 @@ func createAgentOrDie(opt *options.CollectorRunOptions, cfg *configuration.Confi
 	}
 
 	// create source and sink managers
-	sourceManager := createSourceManagerOrDie(opt.Sources, opt.ScrapeTimeout)
+	sourceManager := sources.NewSourceManager(opt.Sources)
```

Comment:
Have NewSourceManager return an error if no sources are provided or if after parsing, the number of valid sources is 0? Let's retain the createSourceManagerOrDie(...) method and do error handling within that.

Reply:
Why? Mow the config is dynamic, maybe just a log message to alert the user ?

Reply:
I think it's a fatal situation if all sources are invalid. Users will not be checking the logs most of the time. Good to fail fast in this situation.